### DISCUSSION TITLE
Discuss user requirements for Git on z/OS

### DISCUSSION THREAD

**Discussion by IgorTodorovskiIBM** [zopencommunity/meta#40] [General]
* Should Git on z/OS tag all files that are cloned
* ...

**Comment by khickel** [comment]
I would like to see an option/env var to control this behavior.  Rocket git requires that the repo contain a .gitattributes file, which is fine for z/os aware repositories, but if you're cloning some general open source repo, this file wont exist (or at least won't have the relevant setting). Then you have to do a bare clone, create the gitattributes file in the .git/info directory and then do the checkout, which is "annoying".
So, having a controllable option that says, "hey, this repo I'm cloning doesn't know about tags, just tag everything as ascii/utf-8/whatever", would be quite handy.  It will still cause problems, because some repos contain binary files, so now you need a way to de-tag those and keep them that way. Nothing is simple.

**Reply by MikeFultonDev** [reply]
Yes!

**Comment by IgorTodorovskiIBM** [comment]
I added some thoughts here with respect to supporting working tree encodings in the .gitattributes: https://github.com/ZOSOpenTools/gitport/issues/21 .  In short, I think we should leverage the upstream working-tree-encoding support and enhance it by adding support for an OS flag to apply to it select platforms.

**Comment by MikeFultonDev** [comment]
I am closing this @IgorTodorovskiIBM - reopen @IgorTodorovskiIBM or @khickel if appropriate. 

Details of working tree encodings are discussed here: https://zosopentools.github.io/meta/#/Guides/GitOnZOS
